src/lib/contracts.ts

```diff
@@ -51,13 +94,30 @@ function isValidRequirementType(
 	return requirement in requirementVersions;
 }
 
+function partition<T>(arr: T[], predicate: (t: T) => boolean) {
```

Comment:
`_.partition` has the same functionality as this, are we trying to stay away from lodash though?

Reply:
yes, ideally we should phase out lodash usage